### DISCUSSION TITLE
Allow migration scripts to distinguish between networks when using Dashboard

### DISCUSSION THREAD

**Discussion by aviaisenberg** [ConsenSys-archive/truffle#4836] [Ideas]
Ran into an issue when writing automated deployment scripts that were different based on the network I was using. 

Noticed that when using dashboard, the network information is stored correct in the build files, but the network name is passed as dashboard to the migration script. 

It would be really useful if the dashboard network would also convey which network it was connected to when being passed into the migration scripts.

**Comment by gnidan** [comment]
This is unfortunate, but I think for the time being, the best option is to use some other identifier, such as network ID or chain ID.

Alternatively, you could set up separate networks yourself by configuring networks in truffle-config.js the old-fashioned way, specifying `url: "http://localhost:24012/rpc"` to indicate that that network should connect to Truffle Dashboard.

(Worth moving this to "issues" as a new feature enhancement request!)

**Reply by aviaisenberg** [reply]
First, thanks for getting back so quickly! Specifying the URL is a great suggestion and Ill go that route for now. 

How do you get the network/chain id into the migration?

**Reply by gnidan** [reply]
> Specifying the URL is a great suggestion and Ill go that route for now.

Glad this suits you!

> How do you get the network/chain id into the migration?

I think the only way is by fetching it yourself with a manual provider request or some other means. Sorry :(
